(DOCSP-23435): Reserved File Names Warning (#74)
mongodb/docs-app-services: source/manage-apps/configure/config.txt

- ``.js`` files define serverless application logic used in functions, triggers,
  HTTPS endpoints, and custom resolvers.

.. important:: Reserved File Names
   
   You can include arbitrarily named files in your realm app config, with a few exceptions.
   The following file names should not be used: ``stitch.json``, ``config.json``, 
   ``realm_config.json``, and ``app_config.json``. Files with these names

>    conflict with pre-made files of the same name in their app structure's file

Reviewer: well it's not that they shouldn't be used, it's that these are reserved file names, so they should be used only for these reserved scenarios. 

and the files aren't only 'pre-made'. you can make them following the structure specified in the app configuration reference - https://www.mongodb.com/docs/atlas/app-services/manage-apps/configure/config/

>    system. Using these file names throws an error when you import the app using the CLI.

Reviewer: doesn't always throw error, only if used in a manner differently from the configuration's format.

- ``.js`` files define serverless application logic used in functions, triggers,
  HTTPS endpoints, and custom resolvers.

.. important:: Reserved File Names
   
   From a CLI perspective, there are a few special/reserved file names we advise

>    you to avoid using for your own purposes: ``stitch.json``, ``config.json``, 

Reviewer: can you phrase this more directly? 

you want to convey the following ideas here i think:
* you can include arbitrarily named files in your realm app config
* with the exception of these specific file names, as they are reserved for config files
* including improperly formatted files with these names causes the Realm CLI to throw an error when you import them

>    ``realm_config.json``, and ``app_config.json``. Files with these names may

Reviewer: ```suggestion
   ``realm_config.json``, and ``app_config.json``. Files with these names
```

not may, they _do_ conflict!

   conflict with pre-made files of the same name in their app structure's file
   system. As such, using these names for files may raise CLI errors.
   

> 

Reviewer: ```suggestion
```

- ``.js`` files define serverless application logic used in functions, triggers,
  HTTPS endpoints, and custom resolvers.

.. important:: Reserved File Names
   
   You can include arbitrarily named files in your realm app config, with a few exceptions.
   Note the following names: ``stitch.json``, ``config.json``, 

>    ``realm_config.json``, and ``app_config.json``. These names are reserved specific 

Reviewer: ```suggestion
   ``realm_config.json``, and ``app_config.json``. These names are reserved for specific 
```

>    files used to define and configure specific components in your app. As such, files 

Reviewer: ```suggestion
   files used to define and configure components in your app. As such, files 
```

>    Note the following names: ``stitch.json``, ``config.json``, 
>    ``realm_config.json``, and ``app_config.json``. These names are reserved for specific 
>    files used to define and configure components in your App. Files 
>    with these names must conform to a specific schema for its component type. 

Reviewer: ```suggestion
   The following filenames are used to define and configure components in your App:
   ``stitch.json``, ``config.json``, ``realm_config.json``, and ``app_config.json``.
   Files with these names must conform to a specific schema for its component type. 
```

   Using these file names when the file's content fails to conform to that schema
  throws an error when you import the app using the CLI.

>    

Reviewer: ```suggestion
   
   Refer to this section for reference information about configuring these files.
```

>    will throw an error when you import the app using the CLI.

Reviewer: ```suggestion
  throws an error when you import the app using the CLI.
```

>    

Reviewer: how about another sentence here w a link to where the user can learn more abt configuring these files? i think this page would be a good place to link to - https://www.mongodb.com/docs/atlas/app-services/manage-apps/configure/config/

Author: i agree about having a reference, but isn't the link you sent just the same page we're editing now?

- ``.js`` files define serverless application logic used in functions, triggers,
  HTTPS endpoints, and custom resolvers.

.. important:: Reserved File Names
   
   You can include arbitrarily named files in your realm app config, with a few exceptions.
   The following filenames are used to define and configure components in your App:
   ``stitch.json``, ``config.json``, ``realm_config.json``, and ``app_config.json``.

>    Files with these names must conform to a specific schema for its component type. 

Reviewer: ```suggestion
   Files with these names must conform to a specific schema for their component type. 
```
